activejob/test/integration/queuing_test.rb

```diff
@@ -145,4 +146,12 @@ class QueuingTest < ActiveSupport::TestCase
     assert job_executed "#{@id}.2"
     assert job_executed_at("#{@id}.2") < job_executed_at("#{@id}.1")
   end
+
+  test "inline jobs run on separate threads" do
+    skip unless adapter_is?(:inline)
+
+    assert_nil Thread.current[:job_ran]
+    ThreadJob.perform_later
+    assert_nil Thread.current[:job_ran]
```

Comment:
I think this test needs 2 threads to test that threading is working correctly. Otherwise you just have one thread and it will always pass regardless of the Thread.new block

Reply:
I agree this test could be better. I tried inspecting the thread spun up by the inline adapter, but `Thread.list` doesn't show it because it dies immediately after joining. How can I use two threads to make this test better?

Reply:
That should do it

```diff
diff --git a/activejob/test/integration/queuing_test.rb b/activejob/test/integration/queuing_test.rb
index 7c92496331..a7d3865c42 100644
--- a/activejob/test/integration/queuing_test.rb
+++ b/activejob/test/integration/queuing_test.rb
@@ -150,8 +150,15 @@ class QueuingTest < ActiveSupport::TestCase
   test "inline jobs run on separate threads" do
     skip unless adapter_is?(:inline)

-    assert_nil Thread.current[:job_ran]
+    t1 = Thread.new do
+      ThreadJob::LatchManager.latch1.wait
+      assert_nil Thread.current[:job_ran]
+      assert(ThreadJob::LatchManager.job_thread[:job_ran])
+      ThreadJob::LatchManager.latch2.count_down
+    end
+
     ThreadJob.perform_later
+    t1.join
     assert_nil Thread.current[:job_ran]
   end
 end
diff --git a/activejob/test/jobs/thread_job.rb b/activejob/test/jobs/thread_job.rb
index 804d18b4a9..27218760a0 100644
--- a/activejob/test/jobs/thread_job.rb
+++ b/activejob/test/jobs/thread_job.rb
@@ -1,7 +1,24 @@
 # frozen_string_literal: true

 class ThreadJob < ActiveJob::Base
+  class LatchManager
+    class << self
+      attr_accessor :job_thread
+
+      def latch1
+        @@latch1 ||= Concurrent::CountDownLatch.new
+      end
+
+      def latch2
+        @@latch2 ||= Concurrent::CountDownLatch.new
+      end
+    end
+  end
+
   def perform
     Thread.current[:job_ran] = true
+    LatchManager.job_thread = Thread.current
+    LatchManager.latch1.count_down
+    LatchManager.latch2.wait
   end
 end
```